src/dmd/backend/elfobj.d

```diff
@@ -1635,8 +1634,9 @@ void Obj_compiler()
     comment_data = cast(Outbuffer*) calloc(1, Outbuffer.sizeof);
     assert(comment_data);
 
-
-    comment_data.write(.compiler,(.compiler).sizeof);
+    char[20 + 8 /* v2.0XX.X */] compiler = void;
```

Comment:
Same issue here, the buffer might not be large enough. I recommend skipping `sprintf` and instead just use D strings and concatenated them with a built-in operator `~`.

Reply:
string concatenation didn't work for me because it requires the GC with -betterC.
I'll use `strcat`, there will be a better way though..

Reply:
If the buffer isn't large enough, just `assert`. This would be acceptable because the compiler version isn't user supplied.